ZHA common error codes in ZHA logs and fixes (#33421)
home-assistant/home-assistant.github.io: source/_integrations/zha.markdown

    network_mode: host
```

### Common error codes in ZHA logs and other log messages

#### EZSP error - NCP entered failed state

> When you see `NCP entered failed state. Requesting APP controller restart` in ZHA logs during normal operation, that indicates was or is a drop in communication between ZHA and the serial interface of the Silabs EmberZNet Zigbee Coordinator (EZSP).

Reviewer: Wording such as "that indicates was or is a drop in" is extremely unclear and the time period for when this happened is not relevant. Let's remove unnecessary text wherever possible:

```suggestion
When you see `NCP entered failed state. Requesting APP controller restart` in ZHA logs during normal operation,  there was drop in communication between ZHA and the coordinator.
```

> The root cause is lost serial connection and correlated symptoms for reset connections include freezes and device non-responsiveness. The solution is to make sure to always have a stable end-to-end serial connection to the Zigbee Coordinator adapter.

Reviewer: Nowadays few people use WiFi coordinators. The root cause is 99% of the time a bad addon or a custom integration locking up the Home Assistant event loop or affecting system stability in some way for ~10 seconds.

Author: Ok, though it unfortunate still common with WiFi Zigbee Coordinator adapters and even the latest models of network-attached Zigbee Coordinator adapters do support using Wi-Fi so some users do use that on them.

The reason is EZSP (EmberZNet Serial Protocol) interface used by Silicon Labs Zigbee Coordinator adapter via bellows requires a stable connection to the serial port, (it is therefore not recommended to use a connection over Wi-Fi, WAN, VPN, etc.).

#### MAC CHANNEL ACCESS FAILURE

The radio error code `MAC_CHANNEL_ACCESS_FAILURE` is common in ZHA logs and means the radio refuses to transmit. This happens when the wireless spectrum is too occupied and that mostly occurs when there is too much EMI/RMI interference, like from EMF sources or when a WiFi network uses the same radio frequency range as the Zigbee channel your Zigbee network in ZHA uses.

If see this often then resolve it by following the suggested actions and tips in the above section about Zigbee interference avoidance and network range/coverage optimization, as well as consider changing either to a different Zigbee channel in ZHA or better yet, try changing your WiFi access point(s) to non-overlapping WiFi channels.

##### NWK TABLE FULL

> `NWK_TABLE_FULL` has the same root cause and solutions as the above `MAC_CHANNEL_ACCESS_FAILURE` error.

Reviewer: Is there a source for this? It's Texas Instruments-specific and not documented anywhere.

Author: Source is the Zigbee2MQTT FAQ page
